controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -721,6 +724,31 @@ private boolean isSubscriberToNonSubscriberTransition(final ReaderGroupConfigRec
            ReaderGroupState.DELETING, currentState, context, executor));
     }
 
+    /**
+     * Delete Scope Recursively.
+     *
+     * @param scope           scope to be deleted.
+     * @param requestId       request id.
+     * @return deletion status.
+     */
+    public CompletableFuture<DeleteScopeStatus.Status> deleteScopeRecursive(final String scope,
+                                                                              final long requestId) {
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        return RetryHelper.withRetriesAsync(() -> streamMetadataStore.checkScopeInDeletingTable(scope, context, executor)
+                .thenCompose(exists -> {
+                    if (exists) {
+                        log.debug(requestId, "Scope {} delete is already in progress", scope);
+                        return CompletableFuture.completedFuture(DeleteScopeStatus.Status.FAILURE);
+                    } else {
+                        DeleteScopeEvent deleteEvent = new DeleteScopeEvent(scope, requestId);
+                        return eventHelper.addIndexAndSubmitTask(deleteEvent, () -> streamMetadataStore.addEntryToDeletingScope(scope, context, executor))
```

Comment:
Before posting this event it may help to check if the scope exists using `checkScopeExists()` and post the event only if scope does not exist.

Reply:
Did you mean `post the event only if scope exists`? Earlier the same check was applied but the code was changed after review. Ref:- https://github.com/pravega/pravega/pull/6312#discussion_r763852610